src/ckb-daemon/usb_linux.c

```diff
@@ -69,13 +69,15 @@ int os_usbsend(usbdevice* kb, const uchar* out_msg, int is_recv, const char* fil
     int res;
     if ((kb->fwversion >= 0x120 || IS_V2_OVERRIDE(kb)) && !is_recv){
         struct usbdevfs_bulktransfer transfer = {0};
-        // FW 2.XX uses 0x03, FW 3.XX uses 0x02
-        transfer.ep = (kb->fwversion >= 0x130 && kb->fwversion < 0x200) ? 4 : (kb->fwversion >= 0x300 ? 2 : 3);
+        // All firmware versions for normal HID devices have the OUT endpoint at the end.
+        // Devices with no input, such as the Polaris, have it at the start.
+        transfer.ep = kb->epcount - IS_SINGLE_EP(kb);
         transfer.len = MSG_SIZE;
         transfer.timeout = 5000;
         transfer.data = (void*)out_msg;
         res = ioctl(kb->handle - 1, USBDEVFS_BULK, &transfer);
     } else {
+        // Note, Ctrl Transfers require an index, not an endpoint, which is why kb->epcount - 1 works
         struct usbdevfs_ctrltransfer transfer = { 0x21, 0x09, 0x0200, kb->epcount - 1, MSG_SIZE, 5000, (void*)out_msg };
```

Comment:
At some point we should refactor this to either use the same style of initialisation as the interrupt struct does, or otherwise use the GNU dot-initialisation method.

Reply:
I think this should wait after the interrupt read branch is merged, to avoid conflicts.